### PR TITLE
Pull request for feature/therapeutic_procedures

### PR DESCRIPTION
**Pull request by Ravennod**
… display of list of therapeutic procedures  at child card's main page.

### CODE REVIEW

**Comment by Ravennod** [comment]
procedure1
procedure2

**Comment by AndriiKatsubo** [comment]
Doesn't work in Firefox.
default

**Review by AndriiKatsubo — DISMISSED** [review]
Changes according to comment message were requested.

**Comment by Ravennod** [comment]
It is working in Firefox now.
firefox_working

**Review by KostetskyiYaroslav — DISMISSED** [review]

**Comment by AndriiKatsubo** [comment]
Input field for date is shifted when compared to other input fields.
Also, I think that this field should have glyphicon too, like other input fields in form.

**Review by AndriiKatsubo — DISMISSED** [review]
Changes were requested according to comment.

**Comment by Ravennod** [comment]
It can't have glyphicon, because it is not standard input element, I've tried to but it looks messed up. Furthermore it looks shifted because other fields have glyphicon.

**Comment by AndriiKatsubo** [comment]
> It can't have glyphicon, because it is not standard input element, I've tried to but it looks messed up. Furthermore it looks shifted because other fields have glyphicon.

I made something similar during working on feature "view_patient_data".
default
Have you looked there?

**Comment by Ravennod** [comment]
Yes, I have, your custom datepicker uses standard input field so you can add glyphicon to it. But I removed datepicker shift.
datepickershift

**Comment by AndriiKatsubo** [comment]
This branch is not in working state
default

**Comment by Shooter75** [comment]
Wait a minute I'll check.

**Review by KostetskyiYaroslav — COMMENTED** [review]
I think after adding of procedure you should to reset form.

**Comment by AndriiKatsubo** [comment]
I think that values like this should be checked.
default
Also, there is horizontal scroll.

**Review by AndriiKatsubo — DISMISSED** [review]
Changes were requested according to comment message.

**Comment by KostetskyiYaroslav** [comment]
I've cloned your branch in a separated folder and I've got an error.
Check migrations that you have created.

**Comment by KostetskyiYaroslav** [comment]
image
Therapeutic Procedure's date should be more that today

File: MedicalDocManagment.WebUI/app/components/childrens-card/child-card/therapeutic-procedures/procedures-add.component.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                            Необхідно зазначити дату призначення
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div class="row">
+                <div class="form-group">
+                    <label for="commentary" class="control-label col-xs-4">Коментар до призначення</label>
+                    <div class="col-xs-4">
+                        <textarea style="min-width: 100%; max-width: 100%;"
```
Comment (KostetskyiYaroslav) [inline MedicalDocManagment.WebUI/app/components/childrens-card/child-card/therapeutic-procedures/procedures-add.component.html]: Add `form-control` class to the `textarea`

**Comment by KostetskyiYaroslav** [comment]
Would be better to sort list of procedures by date.

**Comment by AndriiKatsubo** [comment]
default
I think that number of procedures should be limited to maximum of 999, because that number fits into field from paper form.
default

**Review by AndriiKatsubo — DISMISSED** [review]
Changes were requested according to message about number of procedures.

**Comment by AndriiKatsubo** [comment]
Feature doesn't look like it's paper analogue. Is it done as it should be done?
default

**Comment by KostetskyiYaroslav** [comment]
I think for people it won't be a problem to write 2147483647 down into this field.
image

I can't restore any package from NuGet. I've updated my Windows 10 to last updates and got an errors. 
But I saw code that you have changed. It absolutely suits me.

**Review by KostetskyiYaroslav — APPROVED** [review]

**Comment by AndriiKatsubo** [comment]
> First of all, nobody will ever prescribe 999 procedures, so we can only guess what maximum number of procedures should be. In my opinion, if somebody is mad enough to prescribe 999 we should give him a chance to prescribe 2,147,483,647. 

Still, I disagree with your 2,147,483,647. If you want to get an approve now then change to 999 or some other rational number, otherwise you'll have to wait till this conflict will be resolved with somebody more authoritive.

> And talking about paper analogue, it is not a doctor's specific form, so it shouldn't look like that. Considering my conversation with our mentor, I realize that this feature should be done the way it is already done.

Did you talk with our trainer about your vision of this feature and did he approve the way you see it?

Why did you dismiss my review? Changes, which I requested, were not done.

**Review by AndriiKatsubo — DISMISSED** [review]
Changes were requested according to previously dismissed review.

**Comment by AndriiKatsubo** [comment]
> I think for people it won't be a problem to write 2147483647 down into this field.

That's not rational — usually this form is filled by handwriting, not a printing like you showed.

**Review by AndriiKatsubo — APPROVED** [review]

**Pull request merged**